app/models/work_package/pdf_export/export/markdown.rb

```diff
@@ -94,22 +94,27 @@ def handle_wp_mention_html_tag(tag, node, opts)
         # clear the text content, so it does not get rendered
         next_node.string_content = ""
       end
-      wp_mention_macro(tag.attr("data-text") || "", id, opts)
+      wp_mention_macro(tag.attr("data-text") || "", id.delete("#"), opts)
     end
 
-    def wp_mention_macro(content, id, opts)
+    def expand_wp_mention(work_package, content)
       count = content.count("#")
       if count > 1
-        work_package = WorkPackage.find_by(id: id)
-        unless work_package.nil? || !work_package.visible?
-          # ##1234: {Type} #{ID}: {Subject}
-          content = "#{work_package.type} ##{work_package.id}: #{work_package.subject}"
-          if count == 3
-            # ###1234: {Status} {Type} #{ID}: {Subject} ({Start Date} - {End Date})
-            content = "#{work_package.status.name} #{content}#{work_package_dates(work_package)}"
-          end
+        # ##1234: {Type} #{ID}: {Subject}
+        content = "#{work_package.type} ##{work_package.id}: #{work_package.subject}"
+        if count == 3
+          # ###1234: {Status} {Type} #{ID}: {Subject} ({Start Date} - {End Date})
+          content = "#{work_package.status.name} #{content}#{work_package_dates(work_package)}"
         end
       end
+      content
+    end
+
+    def wp_mention_macro(content, id, opts)
+      work_package = WorkPackage.find_by(id: id)
+      return [text_hash(content, opts)] if work_package.nil? || !work_package.visible?
```

Comment:
Can be more compact:
```suggestion
      return [text_hash(content, opts)] if !work_package&.visible?
```
or
```suggestion
      return [text_hash(content, opts)] unless work_package&.visible?
```

Reply:
Thanks, I like the unless variant.

fixed in https://github.com/opf/openproject/pull/19490/commits/c155ad632ffc38fe73f7cecf2601aebf036086fb